### ISSUE TITLE
URGENT - WHMCS v8 - Logs you out

### ISSUE THREAD

**Issue by qhgh** [gogetssl/whmcs-addon#147]
**Describe the bug**

1. On clicking the 'Manage SSL' button from within WHMCS admin on any SSL, then WHMCS logs you out of the system.

2. On clicking 'Configure SSL' from within the client area it says: 
```
The following errors occurred:
Can not fetch Web Server list, plase refresh page or contact support
```

At the moment we cannot sell any SSL as we cannot configure them as the user or admin.

Suspect this is due to new WHMCS users functionality? https://docs.whmcs.com/Changelog:WHMCS_V8.0

Thanks,
C

**Comment by DennisSkov** [comment]
You're not telling me that you upgraded to WHMCS v8 when this module doesn't support v8 yet?
Either disable the SSLCenter module or downgrade WHMCS.

**Comment by vlad-evg** [comment]
Let us check that and compatibility with v8

**Comment by DennisSkov** [comment]
> Let us check that and compatibility with v8

You're gonna be in for a treat... Clients are no longer login-able entities anymore. They rewrote a lot of the login system to be able to introduce users. Our own modules required a fair bit of rewriting to work with WHMCS v8.

**Comment by qhgh** [comment]
> You're not telling me that you upgraded to WHMCS v8 when this module doesn't support v8 yet?
> Either disable the SSLCenter module or downgrade WHMCS.
> 
> **Always** test on a development installation first.

Noted for next time!

**Comment by qhgh** [comment]
> Let us check that and compatibility with v8

Thank you very much!

**Comment by vlad-evg** [comment]
We are trying to adopt SSL addon to V8 ASAP

**Comment by it2ready** [comment]
When can we expect a version compatible with the 8?

**Comment by DennisSkov** [comment]
Don't rush them.
Also, you shouldn't just update to v8. There's a lot of issues with v8 right now - I wouldn't want to upgrade before these have been fixed.

**Comment by vlad-evg** [comment]
We expect it within 4-5 weeks. We do agree with DennisSkov as v8 is just first release and for sure would have issues

**Comment by alamaxjashek** [comment]
I have the same error, too. And I am wating for LEI Code Number input field in the SSL and the LEI Code module. Do you have any news about this?

**Comment by vlad-evg** [comment]
Modules garden rushing and using all resources to update WHMCS to x8 with all other features.

**Comment by alamaxjashek** [comment]
The module garden zip file is corrupt.. I write to they.  Do you have this
module files? Can you send me by this way?

Thanks!!


Saludos,
*Amilcar Angeles.-*
*Consultor Profesional de TI*
*+1-809-657-7751.*

*Santo Domingo, R.D.*

**Comment by vlad-evg** [comment]
We face no issues with ZIP files, they are generated by Github. Please download it from here:
https://github.com/gogetssl/whmcs-addon/releases/tag/v2.4.10

Again, those files do not support v8 yet

**Comment by ordubilisim** [comment]
Hi whmcs v8.0.3 not working

**Comment by vlad-evg** [comment]
Hi,

Yes, we expect to release SSL addon on 27-28 October.

**Comment by qhgh** [comment]
> Hi,
> 
> Yes, we expect to release SSL addon on 27-28 October.

Excellent news! Thanks 👍

**Comment by vlad-evg** [comment]
WHMCS x8 [Released](https://github.com/gogetssl/whmcs-addon/releases/tag/v2.5.0a)

**Comment by qhgh** [comment]
Thanks for this!

I have tested but am unable to Manage SSL with a OOps Error. It seems this is because the SSL was not created, but there is no Create SSL button? 

![manage-ssl](https://user-images.githubusercontent.com/43718405/97578273-76225280-19e8-11eb-963c-668116d7f6f4.png)
![oops-page-not-found](https://user-images.githubusercontent.com/43718405/97578274-76bae900-19e8-11eb-82dd-4e43206ee101.png)
![cannot-create](https://user-images.githubusercontent.com/43718405/97578275-77537f80-19e8-11eb-8d93-8f9a74506ec5.png)

Thanks for any help!
C

**Comment by vlad-evg** [comment]
Yeap, that is fixed already
https://github.com/gogetssl/whmcs-addon/releases/tag/v2.5.0b

sorry, WHMCS x8 is still unstable and we face issues and can't release a super stable release too

**Comment by qhgh** [comment]
> Yeap, that is fixed already
> https://github.com/gogetssl/whmcs-addon/releases/tag/v2.5.0b
> 
> sorry, WHMCS x8 is still unstable and we face issues and can't release a super stable release too

Thanks for the fix and good work :)

**Comment by qhgh** [comment]
Okay, so we are now importing all the manual orders we made, but it seems it won't show the correct price after import?

Here we see the manual SSL we created £36 and the imported one showing £6.49

![imported-price-wrong-g](https://user-images.githubusercontent.com/43718405/97582528-c5b74d00-19ed-11eb-9a0d-0dc6baa9b0c5.png)

Here we see the correct price is set within the SSL in WHMCS:

![imported-order-whmcs-price-correct-g](https://user-images.githubusercontent.com/43718405/97582531-c6e87a00-19ed-11eb-867b-1a8615fa8b74.png)

But still, it shows £6.49 and not £36?

Any ideas? 

Thanks,
C

**Comment by vlad-evg** [comment]
the modules Garden replied:

"I have checked it many times, the module in the import it does not issue invoices and does not set the price. The price is as shown in the product settings."

:/

**Comment by qhgh** [comment]
Hi Vlad, 

I've checked again just now and you are right, no invoice or emails are sent now 👍 

However, I can see the issue is with **One Time** vs **Annual** Billing Cycles.

If we sent an imported SSL to Annual, it shows the price set on the WHMCS product page. When we set it back to the correct Onetime, it shows the wrong value. This wrong value is not set anywhere on the SSL product page. 

Can you please help on this? 

It's only affecting imported SSL, so it must be an issue there.

Thanks again!
C

**Comment by vlad-evg** [comment]
Thank you for your comment, checking

**Comment by vlad-evg** [comment]
MG's reply:

Sorry, I still don't understand where the issue is, the prices in the screenshots are the same.

Let me describe how the import works

1. The module gets data about the certificate
2. The module determines the billing cycle of the certificate
3. The module creates an order
4. The module does not set the price for the product
5. When creating a new order, only the billing cycle is given and on this basis WHMCS sets the price for the product

As for the price difference between import and standard order, it may result from the commission. The commission is not taken into account when importing.

**Comment by qhgh** [comment]
Hi Vlad,

Thanks for the reply. 

> the prices in the screenshots are the same.

They are not the same. One says £6.49 and one says £36. I've attached the screenshot again:

![importedvsmanually](https://user-images.githubusercontent.com/43718405/97691270-108c9f80-1a96-11eb-90bd-e8c062628745.png)

Both prices are set to £36 in the SSL product page within WHMCS.

new order:

![importedvsnew-new](https://user-images.githubusercontent.com/43718405/97691316-20a47f00-1a96-11eb-934f-b6716fcd4ae8.png)

imported order:

![importedvsnew-imported](https://user-images.githubusercontent.com/43718405/97691331-27cb8d00-1a96-11eb-9e23-333a455783e6.png)

As we can see, both SSL are set to £36. But, the imported SSL shows £6.49 in WHMCS and to the customer. 

PS we do not use commission settings etc. and this only happens on imported SSL.

Does this help to clarify the issue? Are you saying this is a WHMCS issue? I'm happy to raise this with them, but as it only happens to imported SSL, I think its a GGSSL issue?

Thanks,
C

**Comment by vlad-evg** [comment]
Issue solved with [release 2.0.5d](https://github.com/gogetssl/whmcs-addon/releases/tag/v2.5.0d)
